rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -0,0 +1,294 @@
+
+    @Test
+    void test_getFirstInputSigHash_from_multisig() {
+        // Arrange
+        List<BtcECKey> signers = Arrays.asList(
+            BtcECKey.fromPrivate(Hex.decode("fa01")),
+            BtcECKey.fromPrivate(Hex.decode("fa02")),
+            BtcECKey.fromPrivate(Hex.decode("fa03"))
+        );
+
+        List<BtcECKey> pubKeys = signers.stream().map(BtcECKey::getPubKeyPoint).map(BtcECKey::fromPublicOnly)
+            .collect(Collectors.toList());
+        signers.sort(BtcECKey.PUBKEY_COMPARATOR);
+
+        int totalSigners = signers.size();
+        int requiredSignatures = totalSigners / 2 + 1;
+
+        Script redeemScript = ScriptBuilder.createRedeemScript(requiredSignatures, signers);
+
+        Address destinationAddress = BitcoinTestUtils.createP2PKHAddress(btcMainnetParams, "destinationAddress");
+
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams);
+        btcTx.addInput(BitcoinTestUtils.createHash(1), 0, new Script(new byte[]{}));
+        btcTx.addOutput(Coin.COIN, destinationAddress);
+
+        RedeemData redeemData = RedeemData.of(pubKeys, redeemScript);
+        Script p2SHScript = ScriptBuilder.createP2SHOutputScript(requiredSignatures, pubKeys);
+        Script inputScript = p2SHScript.createEmptyInputScript(redeemData.keys.get(0), redeemData.redeemScript);
+        btcTx.getInput(0).setScriptSig(inputScript);
+
+        Sha256Hash expectedSigHash = btcTx.hashForSignature(
+            0,
+            redeemScript,
+            BtcTransaction.SigHash.ALL,
+            false
+        );
+
+        // Act
+        Optional<Sha256Hash> sigHash = BitcoinUtils.getFirstInputSigHash(btcTx);
+
+        // Assert
+        Assertions.assertTrue(sigHash.isPresent());
+        Assertions.assertEquals(expectedSigHash, sigHash.get());
+    }
+
+    @Test
+    void test_getFirstInputSigHash_from_fed() {
+        // Arrange
+        List<BtcECKey> signers = Arrays.asList(
+            BtcECKey.fromPrivate(Hex.decode("fa01")),
+            BtcECKey.fromPrivate(Hex.decode("fa02")),
+            BtcECKey.fromPrivate(Hex.decode("fa03"))
+        );
+        signers.sort(BtcECKey.PUBKEY_COMPARATOR);
+
+        Federation federation = new Federation(
+            FederationTestUtils.getFederationMembersWithBtcKeys(signers),
+            Instant.ofEpochMilli(1000L),
+            0L,
+            btcMainnetParams
+        );
+
+        Address destinationAddress = BitcoinTestUtils.createP2PKHAddress(btcMainnetParams, "destinationAddress");
+
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams);
+        btcTx.addInput(BitcoinTestUtils.createHash(1), 0, new Script(new byte[]{}));
+        btcTx.addOutput(Coin.COIN, destinationAddress);
+
+        Script fedInputScript = PegTestUtils.createBaseInputScriptThatSpendsFromTheFederation(federation);
+        btcTx.getInput(0).setScriptSig(fedInputScript);
+
+        Sha256Hash expectedSigHash = btcTx.hashForSignature(
+            0,
```

Comment:
```suggestion
            FIRST_INPUT_INDEX,
```
We Can Make This A Constant And Use It Everywhere